htdocs/ticket/card.php

```diff
@@ -1000,6 +1001,33 @@
 			}
 		}
 
+		// Contract
+		if (isModEnabled('contrat')) {
+			$langs->load('contracts');
+			$morehtmlref .=  '<br>';
+
+			$morehtmlref .=  $langs->trans('Contract');
+
+			if ($action != 'contrat') {
+				$morehtmlref .= '<a class="editfielda" href="'.$_SERVER["PHP_SELF"].'?action=contrat&amp;id='.$object->id.'">';
```

Comment:
Can you rename the action = 'contrat'  into  'setcontract'  so CSRF security will be effective ?
PS: You will have to add also param token=newToken()...